src/controller_main.c

```diff
@@ -630,38 +599,26 @@ void desktop_listener_task(lv_task_t* data)
                     uint32_t coin_index = BYTE_ARRAY_TO_UINT32(receive_transaction_data.coin_index);
-
-                    get_account_tag(path, receive_transaction_data.address_tag, account_tag, sizeof(account_tag));
+                    const char *token_symbol = get_coin_symbol(coin_index, receive_transaction_data.network_chain_id);
+                    const char *chain_name = get_coin_name(coin_index, receive_transaction_data.network_chain_id);
 
                     if (coin_index == NEAR_COIN_INDEX && receive_transaction_data.near_account_type == 1) {
                         memcpy(&receive_transaction_data.near_registered_account, data_array + offset, 65);
                     }
 
                     if (coin_index == ETHEREUM) {
                         flow_level.level_two = LEVEL_THREE_RECEIVE_TRANSACTION_ETH;
-                        snprintf(flow_level.confirmation_screen_text, sizeof(flow_level.confirmation_screen_text),
-                                 "Receive %s on %s %s in %s", receive_transaction_data.token_name,
-                                 get_coin_name(coin_index, receive_transaction_data.network_chain_id), account_tag,
-                                 wallet.wallet_name);
+                        token_symbol = receive_transaction_data.token_name;
                     } else if (coin_index == NEAR_COIN_INDEX) {
                         flow_level.level_two = LEVEL_THREE_RECEIVE_TRANSACTION_NEAR;
-                        snprintf(flow_level.confirmation_screen_text, sizeof(flow_level.confirmation_screen_text),
-                                 UI_TEXT_RECEIVE_PROMPT,
-                                 get_coin_name(coin_index, receive_transaction_data.network_chain_id),
-                                 account_tag, wallet.wallet_name);
                     } else if (coin_index == SOLANA_COIN_INDEX) {
                         flow_level.level_two = LEVEL_THREE_RECEIVE_TRANSACTION_SOLANA;
-                        snprintf(flow_level.confirmation_screen_text, sizeof(flow_level.confirmation_screen_text),
-                                 UI_TEXT_RECEIVE_PROMPT,
-                                 get_coin_name(coin_index, receive_transaction_data.network_chain_id),
-                                 account_tag, wallet.wallet_name);
                     } else {
                         flow_level.level_two = LEVEL_THREE_RECEIVE_TRANSACTION;
-                        snprintf(flow_level.confirmation_screen_text, sizeof(flow_level.confirmation_screen_text),
-                                 UI_TEXT_RECEIVE_PROMPT,
-                                 get_coin_name(coin_index, receive_transaction_data.network_chain_id),
-                                 account_tag, wallet.wallet_name);
                     }
+
+                    snprintf(flow_level.confirmation_screen_text, sizeof(flow_level.confirmation_screen_text),
+                             UI_TEXT_RECEIVE_PROMPT, token_symbol, chain_name, wallet.wallet_name);
```

Comment:
Do we need to have Coin symbol and name in case of Bitcoin and it's forks too?

Reply:
For Chains that do not support multiple tokens/denominations, we can leave the token/denomination name for send and receive txns confirmations.

Reply:
changed